File: 0693/693.cs
public class Solution {
    public int BinaryNumberWithAlternatingBit(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(BinaryNumberWithAlternatingBit(root.left), BinaryNumberWithAlternatingBit(root.right));
    }
}

File: 0693/693.go
func binaryNumberWithAlternatingBit(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := binaryNumberWithAlternatingBit(root.Left)
	r := binaryNumberWithAlternatingBit(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}